crates/sui-core/src/authority_aggregator.rs

```diff
@@ -1213,17 +1221,21 @@ where
         false
     }
 
-    fn handle_transaction_response_with_signed(
+    fn handle_transaction_response(
         &self,
         state: &mut ProcessTransactionState,
-        signed_transaction: VerifiedSignedTransaction,
+        plain_tx: SignedTransaction,
     ) -> SuiResult<Option<ProcessTransactionResult>> {
-        let (data, sig) = signed_transaction.into_inner().into_data_and_sig();
-        match state.tx_signatures.insert(sig) {
-            InsertResult::NotEnoughVotes => Ok(None),
+        match state.tx_signatures.insert(plain_tx.clone()) {
+            InsertResult::NotEnoughVotes { bad_votes } => {
+                self.metrics.invalid_sig_bad_stake.observe(bad_votes as f64);
+                state.non_retryable_stake += bad_votes;
```

Comment:
we made stake aggregator to return bad_vote and here we have the state to increment non_retryable_stake